Panic in checkid function

Buitl from current master branch
```
$ ./proxmox-api-go -debug -insecure checkid
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
main.main()
	/home/user/proxmox-api-go/main.go:256 +0x3cd5
```

If I pass a value, then checkid works as intended.
```
$ ./proxmox-api-go -debug -insecure checkid 120
{"data":null,"errors":{"vmid":"VM 120 already exists"}}

$ ./proxmox-api-go -debug -insecure checkid 121
{"data":"121"}
2021/07/23 15:53:56 Selected ID is free: 121
```

There should be a test for these cases.

--- a/main.go
+++ b/main.go
@@ -314,6 +314,69 @@ func main() {
 		failError(err)
 		fmt.Println(string(version))
 
+	case "getPoolList":
+		pools, err := c.GetPoolList()
+		if err != nil {
+			log.Printf("Error listing pools %+v\n", err)
+			os.Exit(1)
+		}
+		poolList, err := json.Marshal(pools)
+		fmt.Println(string(poolList))
+
+	case "getPoolInfo":
+		if len(flag.Args()) < 2 {
+			log.Printf("Error poolid required")
+			os.Exit(1)
+		}
+		poolid := flag.Args()[1]
+		poolinfo, err := c.GetPoolInfo(poolid)
+		if err != nil {
+			log.Printf("Error getting pool info %+v\n", err)
+			os.Exit(1)
+		}
+		poolList, err := json.Marshal(poolinfo)
+		fmt.Println(string(poolList))
+
+	case "createPool":
+		if len(flag.Args()) < 2 {
+			log.Printf("Error: poolid required")
+			os.Exit(1)
+		}
+		poolid := flag.Args()[1]
+
+		comment := ""
+		if len(flag.Args()) == 3 {
+			comment = flag.Args()[2]
+		}
+
+		err := c.CreatePool(poolid, comment)
+		failError(err)
+		fmt.Printf("Pool %s created\n", poolid)
+
+	case "deletePool":
+		if len(flag.Args()) < 2 {
+			log.Printf("Error: poolid required")
+			os.Exit(1)
+		}
+		poolid := flag.Args()[1]
+
+		err := c.DeletePool(poolid)
+		failError(err)
+		fmt.Printf("Pool %s removed\n", poolid)
+
+	case "updatePoolComment":
+		if len(flag.Args()) < 3 {
+			log.Printf("Error: poolid and comment required")
+			os.Exit(1)
+		}
+
+		poolid := flag.Args()[1]
+		comment := flag.Args()[2]
+
+		err := c.UpdatePoolComment(poolid, comment)
+		failError(err)
+		fmt.Printf("Pool %s updated\n", poolid)
+
 	default:
 		fmt.Printf("unknown action, try start|stop vmid\n")
 	}

--- a/main.go
+++ b/main.go
@@ -7,6 +7,7 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"regexp"
 	"strconv"
 
 	"github.com/Telmate/proxmox-api-go/proxmox"
@@ -23,10 +24,19 @@ func main() {
 	if !*insecure {
 		tlsconf = nil
 	}
-	c, _ := proxmox.NewClient(os.Getenv("PM_API_URL"), nil, tlsconf, *taskTimeout)
-	err := c.Login(os.Getenv("PM_USER"), os.Getenv("PM_PASS"), os.Getenv("PM_OTP"))
-	if err != nil {
-		log.Fatal(err)
+	c, err := proxmox.NewClient(os.Getenv("PM_API_URL"), nil, tlsconf, *taskTimeout)
+	if userRequiresAPIToken(os.Getenv("PM_USER")) {
+		c.SetAPIToken(os.Getenv("PM_USER"), os.Getenv("PM_PASS"))
+		// As test, get the version of the server
+		_, err := c.GetVersion()
+		if err != nil {
+			log.Fatalf("login error: %s", err)
+		}
+	} else {
+		err = c.Login(os.Getenv("PM_USER"), os.Getenv("PM_PASS"), os.Getenv("PM_OTP"))
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
 
 	vmid := *fvmid
@@ -318,4 +328,10 @@ func failError(err error) {
 		log.Fatal(err)
 	}
 	return
+}
+
+var rxUserRequiresToken = regexp.MustCompile("[a-z0-9]+@[a-z0-9]+![a-z0-9]+")
+
+func userRequiresAPIToken(userID string) bool {
+	return rxUserRequiresToken.MatchString(userID)
 }

--- a/main.go
+++ b/main.go
@@ -253,6 +253,10 @@ func main() {
 		log.Printf("Getting Next Free ID: %d\n", id)
 
 	case "checkid":
+		if len(flag.Args()) < 2 {
+			fmt.Printf("Missing vmid\n")
+			os.Exit(1)
+		}
 		i, err := strconv.Atoi(flag.Args()[1])
 		failError(err)
 		id, err := c.VMIdExists(i)

--- a/main.go
+++ b/main.go
@@ -297,6 +297,13 @@ func main() {
 		vmList, err := json.Marshal(vms)
 		fmt.Println(string(vmList))
 
+	case "getVersion":
+		versionInfo, err := c.GetVersion()
+		failError(err)
+		version, err := json.Marshal(versionInfo)
+		failError(err)
+		fmt.Println(string(version))
+
 	default:
 		fmt.Printf("unknown action, try start|stop vmid\n")
 	}

--- a/proxmox/client.go
+++ b/proxmox/client.go
@@ -1363,3 +1363,63 @@ func (c *Client) UpdateVMHA(vmr *VmRef, haState string) (exitStatus interface{},
 
 	return
 }
+
+func (c *Client) GetPoolList() (pools map[string]interface{}, err error) {
+	resp, err := c.session.Get("/pools", nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return ResponseJSON(resp)
+}
+
+func (c *Client) GetPoolInfo(poolid string) (poolInfo map[string]interface{}, err error) {
+	url := fmt.Sprintf("/pools/%s", poolid)
+	resp, err := c.session.Get(url, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return ResponseJSON(resp)
+}
+
+func (c *Client) CreatePool(poolid string, comment string) error {
+	paramMap := map[string]interface{}{
+		"poolid":  poolid,
+		"comment": comment,
+	}
+
+	reqbody := ParamsToBody(paramMap)
+	_, err := c.session.Post("/pools", nil, nil, &reqbody)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) UpdatePoolComment(poolid string, comment string) error {
+	paramMap := map[string]interface{}{
+		"poolid":  poolid,
+		"comment": comment,
+	}
+
+	reqbody := ParamsToBody(paramMap)
+	url := fmt.Sprintf("/pools/%s", poolid)
+	_, err := c.session.Put(url, nil, nil, &reqbody)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) DeletePool(poolid string) error {
+	url := fmt.Sprintf("/pools/%s", poolid)
+	_, err := c.session.Delete(url, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}

--- a/proxmox/client.go
+++ b/proxmox/client.go
@@ -112,6 +112,15 @@ func (c *Client) Login(username string, password string, otp string) (err error)
 	return c.session.Login(username, password, otp)
 }
 
+func (c *Client) GetVersion() (data map[string]interface{}, err error) {
+	resp, err := c.session.Get("/version", nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return ResponseJSON(resp)
+}
+
 func (c *Client) GetJsonRetryable(url string, data *map[string]interface{}, tries int) error {
 	var statErr error
 	for ii := 0; ii < tries; ii++ {

